optuna/trial/_trial.py

```diff
@@ -485,6 +485,14 @@ def objective(trial):
             )
             raise TypeError(message) from None
 
+        try:
+            step = int(step)
```

Comment:
How about a warning if the conversion changes the value? For example, `step = 1.1`.

Reply:
Thanks! That sounds much better. I've added the warning message.